sycl/include/sycl/detail/accessor_impl.hpp

```diff
@@ -79,11 +80,29 @@ class __SYCL_EXPORT AccessorImplHost {
   AccessorImplHost(id<3> Offset, range<3> AccessRange, range<3> MemoryRange,
                    access::mode AccessMode, detail::SYCLMemObjI *SYCLMemObject,
                    int Dims, int ElemSize, int OffsetInBytes = 0,
-                   bool IsSubBuffer = false, bool IsESIMDAcc = false)
+                   bool IsSubBuffer = false, bool IsESIMDAcc = false,
+                   const property_list &PropertyList = {})
       : MOffset(Offset), MAccessRange(AccessRange), MMemoryRange(MemoryRange),
         MAccessMode(AccessMode), MSYCLMemObj(SYCLMemObject), MDims(Dims),
         MElemSize(ElemSize), MOffsetInBytes(OffsetInBytes),
-        MIsSubBuffer(IsSubBuffer), MIsESIMDAcc(IsESIMDAcc) {}
+        MIsSubBuffer(IsSubBuffer), MIsESIMDAcc(IsESIMDAcc),
+        MPropertyList(PropertyList) {}
+  template <typename... Props>
+  AccessorImplHost(
+      id<3> Offset, range<3> AccessRange, range<3> MemoryRange,
+      access::mode AccessMode, detail::SYCLMemObjI *SYCLMemObject, int Dims,
+      int ElemSize, int OffsetInBytes = 0, bool IsSubBuffer = false,
+      bool IsESIMDAcc = false,
+      const ext::oneapi::accessor_property_list<Props...> &PropertyList = {})
+      : MOffset(Offset), MAccessRange(AccessRange), MMemoryRange(MemoryRange),
+        MAccessMode(AccessMode), MSYCLMemObj(SYCLMemObject), MDims(Dims),
+        MElemSize(ElemSize), MOffsetInBytes(OffsetInBytes),
+        MIsSubBuffer(IsSubBuffer), MIsESIMDAcc(IsESIMDAcc) {
+    // TODO: need some proper solution for handling accessor_property_list,
```

Comment:
Can we do the proper solution from the beginning?

Reply:
This would indeed help us a lot with #6619. I think there's quite some overlap here, so if it was possible to store all runtime-properties correctly (ideally without enumerating them explicitly here), that would be very useful. 

Do I understand correctly that the problem here is the conversion from  `accessor_property_list` to `property_list`, resulting from the fact that the [extension proposal](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_accessor_properties.asciidoc#section-4551-properties-interface) only defines the conversion for lists with no compile-time properties? 
> Accessor property lists with no compile-time-constant properties must be convertible to property_lists and vice versa.

Reply:
It is going to be interesting once we want to integrate [sycl_ext_oneapi_properties](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_properties.asciidoc) into accessor.

Reply:
@sommerlukas in general there is no any known conversion from `accessor_property_list` to `property_list`, there is only `property_list` to `accessor_property_list` conversion. I believe I can implement it in this patch.

Reply:
Done in fb9beca0b25db3b31f1f975f522fddf205e9330f

Reply:
@KornevNikita Thanks for implementing this, will be very useful for our use-case.